src/lib/listeners/ProgressEventListener.ts

```diff
@@ -144,12 +153,17 @@ export class EngineRunProgressSpinner extends ProgressSpinner implements Progres
 		this.isListening = false;
 	}
 
-	public listen(codeAnalyzer: CodeAnalyzer): void {
+	public listen(...codeAnalyzers: CodeAnalyzer[]): void {
+		if (codeAnalyzers.length > 1) {
+			// It's technically possible to implement multiple-core support for this spinner, but not currently necessary.
+			// So just throw a Developer Error if anyone tries.
+			throw new Error('Developer Error: EngineRunProgressSpinner cannot listen to multiple Code Analyzers');
```

Comment:
This error should only ever crop up during development when one of us does something foolish, so the message doesn't require polish.